### DISCUSSION TITLE
Does it work on a personal PC ?

### DISCUSSION THREAD

**Discussion by RaoHammas** [Dokploy/dokploy#99] [Q&A]
Hi,
I was wondering if this works on a truly self-hosted server. e.g. WSL and then using a domain to route to the self-hosted instance?
I tried it by running it on WSL and then setting up an A Record in DNS pointing to local ipv 4 of server but my domain was unable to successfully open the hosted dokploy server.

**Comment by Mr1Blaze** [comment]
Yes it's works on WSL but not anytime. You can use an Virtual Box for more powerful and much more efficient for applications

**Comment by Siumauricio** [comment]
Of course, but only that you will probably have to do extra work to make it work.

**Comment by RaoHammas** [comment]
> Of course, but only that you will probably have to do extra work to make it work.

I tried many things, but domain routing didn't work. I have a domain for which i setup the custom DNS rules which pointed to my local DokPloy server address. In server settings of dokploy i added domain as shown in documentation and YouTube tutorial but it just did not work. I checked DNS was properly configured and was available publicly but for some reason when opening the URL it just said BadRequest with a white page...

**Reply by Mr1Blaze** [reply]
Add me on discord and I will help you.

**Comment by Siumauricio** [comment]
As I mentioned it will be a little difficult for you to try to configure all that, why don't you use a vps provider, it's much simpler and comes with all the headaches you will have trying to configure ip routes, system stuff.

But i know there is a way to make it work the way you want it to work, it's just that it will take you a long time to make it work because of what I already mentioned.

I'd love to help you but I'm busy with other things at the moment, maybe in the future I'll take a look at running it on a local machine.
